src/pipelinedefinition.cpp

```diff
@@ -528,13 +542,31 @@ class NodeValidator {
                 return result;
             }
 
+            if (dependantNodeInfo.gatherFromNode) {
+                result = checkIfGatherNodeCorrect(dependantNodeInfo);
+                if (!result.ok()) {
+                    return result;
+                }
+            }
+
             prepareRemainingUnconnectedDependantModelInputsSet();
         }
 
         if (dependantNodeInfo.kind == NodeKind::CUSTOM) {
             if (!dependantNodeInfo.library.isValid()) {
                 return StatusCode::PIPELINE_DEFINITION_INVALID_NODE_LIBRARY;
             }
+
+            if (dependantNodeInfo.gatherFromNode) {
```

Comment:
Is it possible to not duplicate the code and call it once at one level higher? Not in DL/CUSTOM ifs but for every node.